elytron/src/main/java/org/wildfly/extension/elytron/EvidenceDecoderDefinitions.java

```diff
@@ -0,0 +1,147 @@
+import org.wildfly.extension.elytron.TrivialService.ValueSupplier;
+import org.wildfly.security.auth.server.EvidenceDecoder;
+import org.wildfly.security.x500.GeneralName;
+import org.wildfly.security.x500.principal.X500SubjectEvidenceDecoder;
+import org.wildfly.security.x500.principal.X509SubjectAltNameEvidenceDecoder;
+
+/**
+ * Holder for {@link ResourceDefinition} instances for services that return an {@link EvidenceDecoder}.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+class EvidenceDecoderDefinitions {
+
+    private static final String RFC_822_NAME = "rfc822Name";
+    private static final String DNS_NAME = "dNSName";
+    private static final String DIRECTORY_NAME = "directoryName";
+    private static final String URI_NAME = "uniformResourceIdentifier";
+    private static final String IP_ADDRESS = "iPAddress";
+    private static final String REGISTERED_ID = "registeredID";
+
+    private enum SubjectAltNameType {
+
+        RFC_822_NAME_TYPE(RFC_822_NAME, GeneralName.RFC_822_NAME),
+        DNS_NAME_TYPE(DNS_NAME, GeneralName.DNS_NAME),
+        DIRECTORY_NAME_TYPE(DIRECTORY_NAME, GeneralName.DIRECTORY_NAME),
+        URI_NAME_TYPE(URI_NAME, GeneralName.URI_NAME),
+        IP_ADDRESS_TYPE(IP_ADDRESS, GeneralName.IP_ADDRESS),
+        REGISTERED_ID_TYPE(REGISTERED_ID, GeneralName.REGISTERED_ID);
+
+        private final String name;
+        private final int type;
+
+        SubjectAltNameType(final String name, final int type) {
+            this.name = name;
+            this.type = type;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public int getType() {
+            return type;
+        }
+
+        public static SubjectAltNameType fromName(final String name) {
+            switch (name) {
+                case RFC_822_NAME:
+                    return RFC_822_NAME_TYPE;
+                case DNS_NAME:
+                    return DNS_NAME_TYPE;
+                case DIRECTORY_NAME:
+                    return DIRECTORY_NAME_TYPE;
+                case URI_NAME:
+                    return URI_NAME_TYPE;
+                case IP_ADDRESS:
+                    return IP_ADDRESS_TYPE;
+                case REGISTERED_ID:
+                    return REGISTERED_ID_TYPE;
+                default:
+                    throw new IllegalArgumentException(name);
+            }
+        }
+    }
+
+    static final SimpleAttributeDefinition ALT_NAME_TYPE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ALT_NAME_TYPE, ModelType.STRING, false)
+            .setValidator(new StringAllowedValuesValidator(RFC_822_NAME, DNS_NAME, DIRECTORY_NAME, URI_NAME, IP_ADDRESS, REGISTERED_ID))
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final SimpleAttributeDefinition SEGMENT = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.SEGMENT, ModelType.INT, true)
```

Comment:
It could be checked whether the value is non-negative (>=0).

Reply:
Thanks @OndrejKotek, fixed.